### PR TITLE
Migration from ADAL to MSAL

### PR DESCRIPTION
**Pull request by Shama-K**

### CODE REVIEW

**Comment by jmprieur** [comment]
@Shama-K : why are we doing this work?
We already have a sample using MSAL for this scenario: https://github.com/Azure-Samples/ms-identity-aspnet-webapi-onbehalfof ?

Shouldn't we just archive this sample? (I thought it was archived, BTW)

@Shama-K @kalyankrishna1 : am I missing something?

**Comment by kalyankrishna1** [comment]
JM, we just decided to this opportunity to address as many asp.net scenarios as possible. Most of the code developed will go to a new sample, but till we are done, we are using this repo for all the dev and testing

---
In reply to: [759486516](https://github.com/Azure-Samples/active-directory-dotnet-webapi-onbehalfof-ca/pull/16#issuecomment-759486516) [](ancestors = 759486516)

File: README.md [hunk 1]
```diff
@@ -1,331 +1,372 @@
...
+1. Find the key `ida:ClientSecret` and replace the existing value with the key you saved during the creation of `TodoListWebApp-aspnet` copied from the Azure portal.
+1. Find the key `ida:TodoListServiceScope` and replace the existing value with default API scope.
+
+#### Configure Known Client Applications for service (ToDoListService-ASP.NET)
+
+For a middle tier Web API (`ToDoListService-ASP.NET`) to be able to call a downstream Web API, the middle tier app needs to be granted the required permissions as well.
+However, since the middle tier cannot interact with the signed-in user, it needs to be explicitly bound to the client app in its **Azure AD** registration.
+This binding merges the permissions required by both the client and the middle tier Web Api and presents it to the end user in a single consent dialog. The user then consent to this combined set of permissions.
+
+To achieve this, you need to add the **Application Id** of the client app, in the Manifest of the Web API in the `knownClientApplications` property. Here's how:
```
Comment (kalyankrishna1) [inline README.md:285]: >To achieve this, you  [](start = 0, length = 21)

add the step to configure a CA policy as defined here (https://github.com/Azure-Samples/active-directory-dotnet-webapi-onbehalfof-ca/blob/kkrishna/updates2019/README.md#step-4-create-and-link-a-conditional-access-policy) #Resolved

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @Shama-K

**Pull request merged**